homeassistant/components/axis/config_flow.py

```diff
@@ -201,22 +203,42 @@ async def async_step_init(self, user_input=None):
         return await self.async_step_configure_stream()
 
     async def async_step_configure_stream(self, user_input=None):
-        """Manage the Axis device options."""
+        """Manage the Axis device stream options."""
         if user_input is not None:
             self.options.update(user_input)
             return self.async_create_entry(title="", data=self.options)
 
-        profiles = [DEFAULT_STREAM_PROFILE]
-        for profile in self.device.api.vapix.streaming_profiles:
-            profiles.append(profile.name)
+        schema = {}
+
+        # Stream profiles
+
+        if self.device.api.vapix.params.stream_profiles_max_groups > 0:
+
+            stream_profiles = [DEFAULT_STREAM_PROFILE]
+            for profile in self.device.api.vapix.streaming_profiles:
+                stream_profiles.append(profile.name)
+
+            schema[
+                vol.Optional(
+                    CONF_STREAM_PROFILE, default=self.device.option_stream_profile
+                )
+            ] = vol.In(stream_profiles)
+
+        # Video sources
+
+        if self.device.api.vapix.params.image_nbrofviews > 0:
+            await self.device.api.vapix.params.update_image()
```

Comment:
Can this throw an exception ?

Reply:
I think it could possibly throw an exception if the device is offline, but other mechanisms should show the connection as broken, most probably this is also done around the time the integration is set up

Reply:
It would be nice to give the user a better error if the device can't be reached, but it seems unlikely that they would be changing options for an offline device so it probably not worth it.